### PR TITLE
More revised tests

### PR DESCRIPTION
**Pull request by pbetkier**
More tests on the path to delete `EnvoyControlTestConfiguration`.

### CODE REVIEW

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HostHeaderRewritingTest.kt [hunk 1]
```diff
@@ -73,3 +74,10 @@ internal class HostHeaderRewritingTest : EnvoyControlTestConfiguration() {
         }
     }
 }
+
+fun ObjectAssert<Response>.hasHostHeaderWithValue(overriddenHostHeader: String): ObjectAssert<Response> {
```
Comment (slonka) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HostHeaderRewritingTest.kt:78]: `hasHostHeaderWithValue` in `EnvoyControlTestConfiguration` is no longer used anywhere. You can remove it.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt [hunk 2]
```diff
@@ -57,6 +57,7 @@ val AdsWithNoDependencies = EnvoyConfig("envoy/config_ads_no_dependencies.yaml")
 val Xds = EnvoyConfig("envoy/config_xds.yaml")
 val RandomConfigFile = if (Random.nextBoolean()) Ads else Xds
 
+@Deprecated("use extension approach instead, e.g. RetryPolicyTest")
```
Comment (slonka) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt:60]: ❤️

**Review by slonka — CHANGES_REQUESTED** [review]
One small thing, otherwise looks fine.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HostHeaderRewritingTest.kt (hunk 1 above)
Comment (pbetkier) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HostHeaderRewritingTest.kt:78]: it's now done!

**Review by slonka — APPROVED** [review]

**Review by pzmi — APPROVED** [review]

**Pull request merged**